### DISCUSSION TITLE
' Invalid get index "error" (on base: "Dictionary") ' error when calling 'collection.get()' in FireStore

### DISCUSSION THREAD

**Discussion by Rysm** [GodotNuts/GodotFirebase#165] [Q&A] [answered]
When I call

`collection.get("DOCUMENT_ID")`

I receive the following error:

`Invalid get index "error" (on base: "Dictionary")`

on line 154 in "firestore_task.gd"

```
		match action:
			Task.TASK_LIST, Task.TASK_QUERY:
				data = bod[0].error
				emit_signal("error", data.code, 0, data.message)
			_:
				data = bod.error #this line throws the error
				emit_signal("error", data.code, 0, data.message)
	
```

I've also already pulled the latest GodotFirebase changes at the time of this posting

Thanks in advance for any pointers or assistance!

**Comment by WolfgangSenff** [comment]
Do you have the latest? We just merged (as of about 12 hours ago) a fix for this. If it does not fix it, could you let us know? Our tests ran fine but it's possible we missed testing something.

**Reply by WolfgangSenff** [reply]
Sorry, I saw you said you had latest, but it's weird. We explicitly tried to fix that, so wanting to narrow it down more as to why it isn't working.

**Comment by BearDooks** [comment, marked as answer]
@Rysm I just looked into the latest code with our tester and I am able to get a document from a collection correctly. 

I have been testing with our code and I am having issues reproducing the error.

Can you confirm that you have the right collection set and the document you are trying to get exists?

Thanks

Chuck

**Reply by WolfgangSenff** [reply]
Yes ^ - we basically need more reproduction steps, like what the state of your Firestore is (is the doc actually there, are your rules setup correctly, etc.). https://github.com/GodotNuts/GodotFirebase/issues/166 - if you want to add steps for repro, that would be a better place to do it.

**Reply by BearDooks** [reply]
@Rysm just checking in to see if you could provide your code so we can help look into it. The more you can share the better.

Thanks

~ Chuck

**Reply by Rysm** [reply]
Hi everyone, thanks for reaching out!

Here's a larger snippet of the code, which fires when I press a button from the :

```
	if Firebase.Auth.auth.has("idtoken") == false:
		print("Not logged in")
	else:
		print("Database moment")

		#FIRESTORE REPLACEMENT
		var collection : FirestoreCollection = Firebase.Firestore.collection("testmoment")
		collection.get("testusers")
		var document : FirestoreDocument = yield(collection, "get_document")
```

And here's what my Firestore looks like and while it still gives me an error

![FireStore Setup](https://user-images.githubusercontent.com/8741885/112778114-a6523980-8ff8-11eb-9ed4-4fd59bd2b762.png)

I noticed in the other thread that someone else mentioned that they're also getting an issue without having changed any previous working code

**Reply by BearDooks** [reply]
Have you checked the rules in Firestore to make sure you are allowing people to read its contents? 

We are having a small issue with out plugin and Firestore, but the last commit put a band aid on all of that until we can get the real fix in place.

**Reply by Rysm** [reply]
Hm my permissions look to be fine but I'll definitely do more debugging - thanks for all the help and time you've all provided so far!

**Reply by BearDooks** [reply]
@Rysm I ran into this issue last night. No matter what I did I could not get the plugin to stop throwing the error. 

I finally had to remove the plugin from the project and install a fresh copy. I also had to delete the local cache from my machine before running it. Maybe this will work for you?

We also are working on a new version that fixes some issues  with the Firestore part of the plugin. So stay tuned.

We are still tracking down if that is an issue in Godot or our plugin.

**Reply by BearDooks** [reply]
@Rysm we released a new version of the plugin. I might suggest removing the old one and putting the new one in.

**Reply by Rysm** [reply]
Thanks @BearDooks! It worked - thanks a bunch
